docs/recipes/travis_build_stages.md

```diff
@@ -0,0 +1,90 @@
+# Using semantic-release with [Travis build stages](https://docs.travis-ci.com/user/build-stages/)
+
+![Travis Build Stages screenshot](https://cloud.githubusercontent.com/assets/3729517/25229553/0868909c-25d1-11e7-9263-b076fdef9288.gif)
+
+Build stages are a great way to run semantic release compared to running it in `after_success`:
+
+* The release stage can use Node 8, while the other stages can use other versions, even use a completely different language.
+* It allows you to easily release a project that is not even written in JavaScript, without impacting the job configuration of your test jobs in any way.
+* Instead of having to figure out a "build leader" to do the release and poll the Travis API to wait until all jobs passed, Travis can handle only running the release step after the other jobs passed.
+* If the release failed, you can see at a glance that only the release stage failed while your test stage succeded.
+* Travis can take care of not running the whole release stage at all on non-master branches and PRs.
+
+### Annotated example `.travis.yml`
+
+```yml
+# This could even be a different language
+language: node_js
+
+# Define any version matrix you want for testing
+node_js:
+  - '8'
+  - '6'
+
+# This is inherited by all jobs
+cache:
+  directories:
+    - ~/.npm
+
+# This will be inherited in the release stage
+install:
+  - npm install
+
+# This is the script that will only run for your test stage
+script:
+  - npm run build
+  - npm run lint
```

Comment:
Can you instead recommend to lint in a `pretest` script?
This way contributors are less likely to push commits without linting?

Reply:
I think tools like lint-staged and husky are great for catching lint errors before pushing but CI always needs to repeat the steps and it's kinda out of scope for this example. This is just copied from a real project where linting takes a significant amount of time that you don't want to spend on every test run.